### PR TITLE
Add linting

### PR DESCRIPTION
**Pull request by aweidner**
Still need to test this to verify all the changes did not break anything, but this changeset will allow us to add eslint as part of our travis builds.  We will need to configure code climate separately.

### CODE REVIEW

File: src/lichess.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ var makeRequest = (function() {
     }
     processRequest();
     return makeRequest;
-})();
+}());
```
Comment (lakinwecker) [inline src/lichess.js]: I don't understand why this version is preferable to the other one.

File: src/slack.js [hunk 2]
```diff
@@ -164,7 +164,7 @@ function isModerator(message) {
 //------------------------------------------------------------------------------
 function requiresModerator(bot, message, config) {
     return Q.fcall(function() {
-        if (typeof message.league == 'undefined') {
+        if (_.isUndefined(message.league)) {
```
Comment (lakinwecker) [inline src/slack.js]: oooh. Nice. I much prefer this.

**Comment by endrawes0** [comment]
PR looks  good. after testing, we can merge. Are we still planning to replace tests for string equality with Lo-Dash `isEqual`?

File: src/lichess.js (hunk 1 above)
Comment (aweidner) [inline src/lichess.js]: Looking at the rule, it seems that either is the same and this is purely a stylistic choice.  http://eslint.org/docs/2.0.0/rules/wrap-iife.  We can set it to inside, outside, or any.  I can change the rule and revert it if you would rather have outside.  I don't have a preference either way.

**Comment by aweidner** [comment]
Yes, I will be changing anything were string comparisons from an unknown source (one of the third party libraries or user input) are involved to `isEqual`.

**Comment by aweidner** [comment]
One thing to note about the change to `_.isEqual`.  It is used only in the case where both the comparisons are expected to be against string data, and both of the operands are unknown.  In the case where we are comparing two integers, `===` should serve us fine.  Also in the case where we are comparing a string against a known value (something like `playerName === "ghostologist"`), `===` will work also.  The only time we need to use `_.isEqual` is in the case where we have `playerName === otherName` as we don't know either operand, and we could encounter the `new String("") !== ""` case.

The only exception to this rule is where we are comparing explicitly against `""` in which case we must also use`_.isEqual`.

File: src/chesster.js [hunk 3]
```diff
@@ -587,7 +588,7 @@ function(bot, message) {
         results,
         function(err, reversed) {
             if (err) {
-                if (err.indexOf && err.indexOf("Unable to find pairing.") == 0) {
+                if (_.includes(err, "Unable to find pairing.")) {
```
Comment (lakinwecker) [inline src/chesster.js]: This is great! :)

File: src/chesster.js [hunk 4]
```diff
@@ -830,22 +831,22 @@ function validateUserResult(details, result){
         valid: true,
         reason: ""
     };
-    if( details.winner && result.result == "1/2-1/2" ){
+    if( details.winner && result.result === "1/2-1/2" ){
```
Comment (lakinwecker) [inline src/chesster.js]: This is probably fine - but this is a case where I think _.isEqual is probably better.

File: src/league.js [hunk 5]
```diff
@@ -529,15 +528,15 @@ league_attributes = {
...
         return Q.fcall(function() {
-            if (self._teams.length == 0) {
+            if (self._teams.length === 0) {
                 return "The {name} league does not have captains".format({
                     name: self.options.name
                 });
             }
             teams = _.filter(self._teams, function(t) {
-                return t.name.toLowerCase() == teamName.toLowerCase()
+                return t.name.toLowerCase() === teamName.toLowerCase()
```
Comment (lakinwecker) [inline src/league.js]: This is the same case as below, we should use _.isEquals for both or neither. I suspect that toLowerCase() is pretty consistent about what types it returns, so === is probably fine.

File: src/lichess.js (hunk 1 above)
Comment (lakinwecker) [inline src/lichess.js]: Ok.  I don't have a stylistic preference either. Let's leave it the default for eslint. Easier that way.

**Comment by lakinwecker** [comment]
This is great stuff. I just reviewed it again and I'm really happy with it.

Looking forward to seeing this go live.

**Comment by endrawes0** [comment]
I am ready to merge this. Any concerns?

File: src/league.js [hunk 6]
```diff
@@ -281,10 +283,7 @@ league_attributes = {
     // Returns whether someone is a moderator or not.
     //--------------------------------------------------------------------------
     'isModerator': function(name) {
-        var self = this;
-        return _.some(self.options.moderators, function(moderator) {
-            return name == moderator;
-        });
+        return _.includes(this.options.moderator, name);
```
Comment (lakinwecker) [inline src/league.js]: This should be plural: ```this.options.moderators```

Comment (endrawes0) [inline src/league.js]: Fixing this in my PR. #86

**Pull request merged**